modules/service/src/test/scala/lucuma/odb/graphql/DatabaseOperations.scala

```diff
@@ -552,57 +553,68 @@ trait DatabaseOperations { this: OdbSuite =>
         .liftTo[IO]
     }
 
-  def setAllocationAs(
-    user: User,
-    pid: Program.Id,
-    partner: Partner,
-    scienceBand: ScienceBand,
-    duration: TimeSpan,
-  ): IO[Unit] =
-    expect(
+  def readAllocations(
+    topLevelField: String,
+    json:          Json
+  ): IO[Set[(Partner, ScienceBand, BigDecimal)]] =
+    json.hcursor.downFields(topLevelField, "allocations").values.toList.flatten.traverse { obj =>
+      val c = obj.hcursor
+      (for {
+        p <- c.downField("partner").as[Partner]
+        b <- c.downField("scienceBand").as[ScienceBand]
+        d <- c.downFields("duration", "hours").as[BigDecimal]
+      } yield (p, b, d))
+       .leftMap(f => new RuntimeException(f.getMessage))
+       .liftTo[IO]
+    }.map(_.toSet)
```

Comment:
Since I cannot control the order of allocations coming out of the mutation, tests have to compare the set of allocations returned.